db/emoji_reactions.py

```diff
@@ -49,3 +52,86 @@ def get_reactions_for_user(user_id: int, session: sessionmaker) -> list[str]:
         ).all()
 
         return [row[0].emoji for row in results]
+    
+def get_emoji_reaction_delay(session: sessionmaker) -> int:
+    """
+    Get Robomoji delay time 
+
+    Returns:
+        int: Delay time in seconds for Robomojis
+    """
+    with session() as sess:
+        result = sess.query(EmojiReactionDelay).first()
+
+        if result is None:
+            sess.execute(insert(EmojiReactionDelay).values(delay_in_seconds=DEFAULT_EMOJI_REACTION_DELAY)) # Fallback default value of 15 seconds
+            return DEFAULT_EMOJI_REACTION_DELAY
+        
+        return result.delay_in_seconds
+
+def set_emoji_reaction_delay(delay_time: int, session: sessionmaker) -> int:
+    """
+    Set Robomoji delay time 
+
+    Args:
+        delay_time (int): Delay time in seconds for Robomojis
+    """
+    with session() as sess:
+        result = sess.query(EmojiReactionDelay).first()
+
+        if result is None:
+            sess.execute(insert(EmojiReactionDelay).values(delay_in_seconds=delay_time))
+            return delay_time
+
+        existing_row: EmojiReactionDelay = result
+        sess.execute(
+            update(EmojiReactionDelay)
+            .where(EmojiReactionDelay.id == existing_row.id)
+            .values(delay_in_seconds=delay_time)
+        )
+        return delay_time
+    
+def get_emoji_reaction_last_used(user_id: int, session: sessionmaker) -> datetime:
+    """
+    Get DateTime of last Robomoji reaction
+
+    Args:
+        user_id (int): Discord User ID to add reaction to
+
+    Returns:
+        DateTime: DateTime of last Robomoji reaction for given user
+    """
+    with session() as sess:
+        result = sess.execute(
+            select(EmojiReactionTimes).where(EmojiReactionTimes.user_id == user_id)
+        ).first()
+
+        if result is None:
+            sess.execute(insert(EmojiReactionTimes).values(user_id=user_id, last_reacted=datetime.now())) # First usage of Robomoji in new system by given user
+            return datetime.now() - timedelta(minutes=5)
+        
+        emoji_reaction_time: EmojiReactionTimes = result[0]
+        return emoji_reaction_time.last_reacted
```

Comment:
While I can appreciate that there's almost no way a mod is gonna set the reaction to >5minutes, if they did this would mean that a user's first message does not receive a reaction.

Functionality aside, I also prefer for the controllers to handle the defaulting logic if possible so that there aren't many unexpected "side effects" to calling a DB function. Generally functions like `get_[data]` should probably be considered read-only. I wanna be clear, you can probably find examples of me breaking this rule throughout the codebase, which I'd love to fix up LOL

Personally I'd prefer taking the approach where `get_emoji_reaction_last_used` returns an `Optional[datetime]`. This way our controller can explicitly handle the `None` case and always provide a reaction / update the time to `.now()`